packages/web-app-epub-reader/src/App.vue

```diff
@@ -0,0 +1,161 @@
+      </oc-button>
+    </div>
+    <div id="epub-reader" ref="bookContainer" class="oc-flex oc-flex-center" />
+    <div class="oc-flex oc-flex-middle oc-mx-l">
+      <oc-button :disabled="navigateRightDisabled" appearance="raw" @click="navigateRight">
+        <oc-icon name="arrow-right-s" fill-type="line" size="xlarge" />
+      </oc-button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, nextTick, PropType, ref, unref, watch } from 'vue'
+import { Resource } from '@ownclouders/web-client/src/helpers/resource/types'
+import { AppConfigObject, Key, useKeyboardActions } from '@ownclouders/web-pkg'
+import ePub, { Book, NavItem, Rendition } from 'epubjs'
+
+export default defineComponent({
+  name: 'EpubReader',
+  emits: ['close'],
+  props: {
+    applicationConfig: { type: Object as PropType<AppConfigObject>, required: true },
+    currentContent: {
+      type: String,
+      required: true
+    },
+    isReadOnly: { type: Boolean, required: false },
+    resource: { type: Object as PropType<Resource>, required: true }
+  },
+  setup(props, { $emit }) {
+    const keyboardActions = useKeyboardActions()
+    const bookContainer = ref<Element>()
+    const chapters = ref<NavItem[]>([])
+    const currentChapter = ref<NavItem>({})
+    const navigateLeftDisabled = ref(true)
+    const navigateRightDisabled = ref(false)
+    let book: Book
+    let rendition: Rendition
+
+    const navigateLeft = () => {
+      rendition.prev()
+    }
+
+    const navigateRight = () => {
+      rendition.next()
+    }
+
+    const showChapter = (chapter: NavItem) => {
+      rendition.display(chapter.href)
+    }
+
+    keyboardActions.bindKeyAction({ primary: Key.ArrowLeft }, () => navigateLeft())
+    keyboardActions.bindKeyAction({ primary: Key.ArrowRight }, () => navigateRight())
+
+    watch(
+      () => props.currentContent,
+      async () => {
+        await nextTick()
+
+        if (book) {
+          book.destroy()
+        }
+
+        book = ePub(props.currentContent)
+        book.loaded.navigation.then(({ toc }) => {
+          chapters.value = toc
+          currentChapter.value = toc[0]
+        })
+
+        rendition = book.renderTo(unref(bookContainer), {
+          flow: 'paginated',
+          width: 700,
+          height: '100%'
+        })
+        rendition.display()
+        rendition.on('relocated', () => {
+          const locationCfi = rendition.currentLocation().start.cfi
+          const spineItem = book.spine.get(locationCfi)
+          const navItem = book.navigation.get(spineItem.href)
+
+          // Might be sub nav item and therefore undefined
+          if (navItem) {
+            currentChapter.value = navItem
+          }
+
+          navigateLeftDisabled.value = rendition.currentLocation().atStart === true
+          navigateRightDisabled.value = rendition.currentLocation().atEnd === true
+        })
+
+        rendition.on('keydown', (event: Event) => {
```

Comment:
@dschmidt any idea to not do this? we already have keydown listeners, and actually bubbeling the event up would be sufficient